lib/occi/core/renderers/text/locations.rb

```diff
@@ -29,7 +29,15 @@ def render_plain
           # @return [Hash] textual representation of Object for headers
           def render_headers
             return {} if object.empty?
-            { LOCATION_KEY_HEADERS => object.map(&:to_s) }
+            { LOCATION_KEY_HEADERS => location_ary_or_first }
+          end
+
+          protected
+
+          # :nodoc:
+          def location_ary_or_first
+            locations = object.map(&:to_s)
+            locations.count == 1 ? locations.first : locations
```

Comment:
I think you can use method `many?` here.
http://guides.rubyonrails.org/active_support_core_extensions.html#many-questionmark